counsel.el

```diff
@@ -6312,6 +6327,19 @@ Use the presence of a \".git\" file to determine the root."
 Use the presence of a `dir-locals-file' to determine the root."
   (counsel--dominating-file dir-locals-file))
 
+(defun counsel--single-file-root ()
+  "Return the root of a single standalone file.
```

Comment:
Which file?  Wouldn't it be clearer to just say that it returns the buffer's `default-directory` or something along those lines?

Reply:
Redefined:

    (defun counsel--buffer-default-directory ()
      "Return the `default-directory' of the current buffer.

    To avoid catching unrelated history for this one file we locally
    override `counsel-compile-history' and leave the user to fill in their
    preferred build stanza.

    This should be the last function in `counsel-compile-root-functions'
    after giving all the other project root finders a chance."
      (unless (local-variable-p 'counsel-compile-history-match-buffer-file-name)
        (setq counsel-compile-history-match-buffer-file-name t))
      (file-name-directory (or buffer-file-name default-directory)))

Reply:
Much clearer, thanks.